actors/builtin/verifreg/verified_registry_actor.go

```diff
@@ -228,12 +247,27 @@ func (a Actor) RestoreBytes(rt vmr.Runtime, params *RestoreBytesParams) *adt.Emp
 	}
 
 	var st State
+	rt.State().Readonly(&st)
+	if st.RootKey == params.Address {
+		rt.Abortf(exitcode.ErrIllegalArgument, "Cannot restore allowance for Rootkey")
+	}
```

Comment:
Will this ever catch anything?

I think that currently, the market actor sends this method a non-ID address, because it's the same address used to verify a client's signature over a DealProposal.

This may not catch anything, because (I think) RootKey is supposed to be an ID address.

Reply:
On second thought, this should probably be removed. The market actor uses this during its cron tick - an abort here may allow someone to force a cron tick failure.

Reply:
Same goes for the change below.

Reply:
What if instead of removing these, we just removed the "RequireSuccess" condition from the market actor's cron tick?

Reply:
The market is supposed to resolve these, but verified registry implementation violated this for UseBytes: https://github.com/filecoin-project/specs-actors/issues/768. This case, called from cron, can rely on the ID address and so should check and enforce it.

This implicitly means that, unlike multisig parties, we do require deal participants to have instantiated account actors (fyi @arajasek)